model/placeholders/src/main/java/org/eclipse/ditto/model/placeholders/EnforcementFactoryFactory.java

```diff
@@ -29,8 +29,8 @@
      * @param <I> the type from which the input values are resolved
      * @return the new {@link EnforcementFactoryFactory}
      */
-    public static <I> EnforcementFilterFactory<I, String> newEnforcementFilterFactory(final Enforcement enforcement,
-            final Placeholder<I> inputFilter, final Placeholder<String> filterPlaceholderResolver) {
+    private static <O> EnforcementFilterFactory<O, CharSequence > newEnforcementFilterFactory(final Enforcement enforcement,
```

Comment:
Ah, I was confused about I(nput) and (O)utput there because this private method is called from the public method which used O. I'd revert this change back to use I again and change O in the public method to I aswell as currently O defines the type that is required for input.